### PR TITLE
Update contributors component to match refined mocks. #87.

### PR DESCRIPTION
**Pull request by frano-m**
### Reviewers

#87

@MillenniumFalconMechanic 

### Changes
- updated contributors to match refined mocks.
- added `Contributors` stories.
- added `Contributors` stories to `Project` stories.
- updated storybook config alias with `site-config`
- updated section content styles from `span` to `div` element.

### CODE REVIEW

File: explorer/app/components/Project/components/Section/section.styles.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export const SectionContent = styled(Typography)`
   ${({ theme }) => theme.breakpoints.up(BREAKPOINT.TABLET)} {
     padding: 0;
   }
-`;
+` as typeof Typography;
```
Comment (frano-m) [inline explorer/app/components/Project/components/Section/section.styles.ts:38]: Required: see https://mui.com/material-ui/guides/typescript/#complications-with-the-component-prop

File: explorer/app/components/Project/common/projectTransformer.ts [hunk 2]
```diff
@@ -41,3 +82,57 @@ export function getProjectDescription(
...
+
+/**
+ * Map key-value pair of collaborating organizations and corresponding citation.
+ * @param contributors - Project contributor response model return from API.
+ * @returns collaborating organizations keyed by their citation.
+ */
+function getCitationByCollaboratingOrganizations(
+  contributors: ContributorResponse[]
+): Map<string, number> {
+  const setOfCollaboratingOrganizations = new Set(
```
Comment (MillenniumFalconMechanic) [inline explorer/app/components/Project/common/projectTransformer.ts:117]: Simplify naming here to just `collaboratingOrganizations`?

File: explorer/app/components/Project/common/projectTransformer.ts [hunk 3]
```diff
@@ -1,6 +1,11 @@
 // App dependencies
```
Comment (MillenniumFalconMechanic) [inline explorer/app/components/Project/common/projectTransformer.ts:1]: Really solid code here F, thank you!

File: explorer/site-config/hca-dcp/dev/projectViewModelBuilder.ts [hunk 4]
```diff
@@ -36,19 +37,9 @@ export const projectToContacts = (
 
 export const projectsToContributors = (
```
Comment (MillenniumFalconMechanic) [inline explorer/site-config/hca-dcp/dev/projectViewModelBuilder.ts:38]: Looking at the naming convention here; does it make sense to use a `buildX` format here in the view builder, and then a `getX` format in the corresponding transformer? It looks like we have `getProjectContacts` and `getProjectDescription` in the `projectTransformer` already, and maybe the `buildX` terminology makes more sense here in the builder? What are your thoughts?

File: explorer/app/components/Project/common/projectTransformer.ts [hunk 5]
```diff
@@ -15,6 +20,42 @@ export function buildProjectCitationPath(project?: ProjectResponse): string {
   return `explore/projects/${project.projects[0].projectId}`;
 }
 
+/**
+ * Builds project contributors from API response.
+ * @param project - Project response model return from API.
+ * @returns project contributors with their corresponding [organization] citation number.
+ */
+export function buildProjectContributors(
```
Comment (MillenniumFalconMechanic) [inline explorer/app/components/Project/common/projectTransformer.ts:28]: I am seeing a different order of contributors between local and production. Can you take a look at this?

[Local](http://localhost:3000/explore/projects/60ea42e1-af49-42f5-8164-d641fdb696bc):
Screen Shot 2022-06-28 at 12 25 21 PM

[Production](https://data.humancellatlas.org/explore/projects/60ea42e1-af49-42f5-8164-d641fdb696bc):
Screen Shot 2022-06-28 at 12 25 45 PM

Comment (MillenniumFalconMechanic) [reply explorer/app/components/Project/common/projectTransformer.ts:28]: It also looks like the role has fallen off? See `(Principal Investigator)` in the production screenshot above.

File: explorer/app/components/Project/common/projectTransformer.ts (hunk 2 above)
Comment (frano-m) [reply explorer/app/components/Project/common/projectTransformer.ts:117]: I've simplified to `collaboratingOrganizationsSet` as per our discussion.

File: explorer/app/components/Project/common/projectTransformer.ts [hunk 6]
```diff
@@ -41,3 +82,57 @@ export function getProjectDescription(
...
+  );
+}
+
+/**
+ * Formats name from "firstName,middleName,lastName" to "firstName middleName lastName".
+ * @param commaDelimitedName
+ * @returns formatted name "firstName middleName lastName".
+ */
+function formatContributor(commaDelimitedName: string): string {
+  return commaDelimitedName.split(/[ ,]+/).join(" ");
```
[the remark this answers is not in the document]
Comment (frano-m) [reply explorer/app/components/Project/common/projectTransformer.ts:106]: Sure thing!

File: explorer/app/components/Project/common/projectTransformer.ts (hunk 5 above)
Comment (frano-m) [reply explorer/app/components/Project/common/projectTransformer.ts:28]: The API returned lists the contributors in a different order.
Added the role - thanks for picking it up!

**Review by MillenniumFalconMechanic — APPROVED** [review]
LGTM, thanks Fran!

**Pull request merged**